src/news_gui.cpp

```diff
@@ -755,28 +757,37 @@ static void MoveToNextNewsItem()
 /** Delete a news item from the queue */
 static std::list<NewsItem>::iterator DeleteNewsItem(std::list<NewsItem>::iterator ni)
 {
-	if (_forced_news == ni || _current_news == ni) {
-		/* When we're the current news, go to the previous item first;
+	bool updateCurrentNews = (_forced_news == ni || _current_news == ni);
+	bool updateStatusbarNews = (_statusbar_news == ni);
+
+	if (updateCurrentNews) {
+		/* When we're the current news, go to the next older item first;
 		 * we just possibly made that the last news item. */
-		if (_current_news == ni) _current_news = (_current_news == std::begin(_news)) ? std::end(_news) : std::prev(_current_news);
+		if (_current_news == ni) ++_current_news;
+	}
```

Comment:
You're checking `_forced_news` for being the item that's going to be removed, but it is kept as a dangling iterator.
Wouldn't it be better to ensure it's not dangling at any point? Maybe just resetting to `std::end` might be enough here? Even though the later `MoveToNextItem` will also do that.

Reply:
Done